outputter/outputter.go

```diff
@@ -13,55 +13,63 @@ import (
 )
 
 var (
-	eventsWritten int64
-	bytesWritten  int64
+	EventsWritten map[string]int64
+	BytesWritten  map[string]int64
 	lastTS        time.Time
 	rotchan       chan *config.OutputStats
 	gout          [config.MaxOutputThreads]config.Outputter
 )
 
+func init() {
+	EventsWritten = make(map[string]int64)
+	BytesWritten = make(map[string]int64)
+}
+
 // ROT starts the Read Out Thread which will log statistics about what's being output
 // ROT is intended to be started as a goroutine which will log output every c.
 func ROT(c *config.Config) {
 	rotchan = make(chan *config.OutputStats)
 	go readStats()
 
-	lastEventsWritten := eventsWritten
-	lastBytesWritten := bytesWritten
+	lastEventsWritten := make(map[string]int64)
+	lastBytesWritten := make(map[string]int64)
 	var gbday, eventssec, kbytessec float64
 	var tempEW, tempBW int64
 	lastTS = time.Now()
 	for {
 		timer := time.NewTimer(time.Duration(c.Global.ROTInterval) * time.Second)
 		<-timer.C
 		n := time.Now()
-		tempEW = eventsWritten
-		tempBW = bytesWritten
-		eventssec = float64(tempEW-lastEventsWritten) / float64(int(n.Sub(lastTS))/int(time.Second)/c.Global.ROTInterval)
-		kbytessec = float64(tempBW-lastBytesWritten) / float64(int(n.Sub(lastTS))/int(time.Second)/c.Global.ROTInterval) / 1024
-		gbday = (kbytessec * 60 * 60 * 24) / 1024 / 1024
-		log.Infof("Events/Sec: %.2f Kilobytes/Sec: %.2f GB/Day: %.2f", eventssec, kbytessec, gbday)
+		for k := range BytesWritten {
+			tempEW = EventsWritten[k]
+			tempBW = BytesWritten[k]
+			eventssec = float64(tempEW-lastEventsWritten[k]) / float64(int(n.Sub(lastTS))/int(time.Second)/c.Global.ROTInterval)
+			kbytessec = float64(tempBW-lastBytesWritten[k]) / float64(int(n.Sub(lastTS))/int(time.Second)/c.Global.ROTInterval) / 1024
+			gbday = (kbytessec * 60 * 60 * 24) / 1024 / 1024
+			log.Infof("Sample: %s Events/Sec: %.2f Kilobytes/Sec: %.2f GB/Day: %.2f", k, eventssec, kbytessec, gbday)
```

Comment:
I think rather than outputting a line per sample, which could get excessive, we should sum them up and output that.